File: client-go/game_state/action.go
package game_state

type Action struct {
	PlayerId  string `json:"playerId"`
	GameId    int32  `json:"gameId"`
	ActType   string `json:"actType"`
	Direction string `json:"direction"`
}

func NewAction(playerId string, gameId int32, actType string, direction string) *Action {
	return &Action{playerId, gameId, actType, direction}
}

File: server-kotlin/src/main/kotlin/com/reaktor/kafka/moba/model/Action.kt
package com.reaktor.kafka.moba.model

import com.google.gson.Gson

data class Action(
    var playerId: String = "",
    var gameId: Int = 0,
    var actType: String = "",
    var direction: String = ""
) {

    override fun toString(): String {
        return Gson().toJson(this)
    }
}

